### PR TITLE
add SensitiveParameter to sensitive arguments

### PR DESCRIPTION
**Pull request by NicolasCARPi**
This change adds the PHP attribute SensitiveParameter to the secret holding variables.
See: https://www.php.net/manual/en/class.sensitiveparameter

This feature is only available in PHP 8.2, so the minimum php version required has been updated.

Github Actions now use PHP 8.2 and 8.3 for the tests. The checkout action has been updated to v4, too.

Fix issue #118

### CODE REVIEW

**Comment by willpower232** [comment]
I finally merged #103 so hopefully the v3 composer installs will be slightly smaller, not sure if explicitly worth noting in the changelog

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
+
+## Breaking changes
+
+### PHP Version
+
+Version 3.x requires at least PHP 8.2.
+
+## Other changes
+
+* The new PHP attribute SensitiveParameter was added to the code, to prevent accidental leak of secrets in stack traces.
```
[the remark this answers is not in the document]
Comment (NicolasCARPi) [reply CHANGELOG.md]: Done!

**Comment by RobThree** [comment]
> not sure if explicitly worth noting in the changelog

Meh...

I think the `.phpproj` and `.sln` can also be deleted; I don't use VS2022 with [PHP tools](https://www.devsense.com/en) for PHP development anymore.

**Comment by NicolasCARPi** [comment]
> not sure if explicitly worth noting in the changelog

About the changelog, I would suggest going over the commits added after the latest version, and mentioning the changes in this file.

Also, it looks like the PR are merged without squashing the commits, which pollutes the git log with useless commits such as "Fix typo". I would strongly suggest merging PR squashing the commits into 1.

**Comment by RobThree** [comment]
Hey, guys, not sure if I expressed this before (if I haven't, I should've, I've thought this many a times) but I really appreciate your work on this library. Especially the both of you, @NicolasCARPi  and @willpower232 .

Thanks guys! 🙏 👊

File: CHANGELOG.md
Comment (willpower232) [inline CHANGELOG.md:1]: document that removed argument

**Review by willpower232 — COMMENTED** [review]
might as well make this the v3 pr haha

File: CHANGELOG.md
Comment (NicolasCARPi) [reply CHANGELOG.md:1]: I was thinking of making a distinct PR for the changelog, rather than cramming it all in this one.

Comment (willpower232) [reply CHANGELOG.md:1]: fair enough :pray:

**Comment by NicolasCARPi** [comment]
Is there any blocker for the merge?

**Comment by RobThree** [comment]
I merged it, AFAIK it was all ok.

**Comment by willpower232** [comment]
I meant to poke you all yesterday, all good with me

**Pull request merged**